packages/generator-common/src/file-writer/create-file.ts

```diff
@@ -1,9 +1,138 @@
+import { resolve, join, parse } from 'path';
 import { promises } from 'fs';
-import { join } from 'path';
-import { unixEOL, codeBlock, ErrorWithCause } from '@sap-cloud-sdk/util';
+import {
+  codeBlock,
+  createLogger,
+  ErrorWithCause,
+  unixEOL
+} from '@sap-cloud-sdk/util';
+import prettier, {
+  BuiltInParserName,
+  Options as PrettierOptions
+} from 'prettier';
 import { getCopyrightHeader } from '../util';
 
-const { writeFile } = promises;
+const { writeFile, readFile } = promises;
+const logger = createLogger('create-file');
+
+/**
+ * Options for the creation of a file.
+ * @internal
+ */
+export interface CreateFileOptions {
+  /**
+   * Flag to indicate if the file is overwritten.
+   */
+  overwrite?: boolean;
+  /**
+   * Flag to indicate if copy write header is added.
+   */
+  withCopyright?: boolean;
+  /**
+   * Path to the prettier config with respect to the process.cwd().
+   */
+  prettierConfigPath?: string;
+  /**
+   * Flag to indicate if the file is prettified before.
```

Comment:
[req] Maybe we should avoid "prettier" as a verb and instead say "formatted with prettier".